src/basho_bench_worker.erl

```diff
@@ -180,30 +202,58 @@ terminate(_Reason, _State) ->
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
-
-
 %% ====================================================================
 %% Internal functions
 %% ====================================================================
 
 %%
 %% Expand operations list into tuple suitable for weighted, random draw
 %%
-ops_tuple() ->
+ops_tuple(Operations) ->
     F =
         fun({OpTag, Count}) ->
                 lists:duplicate(Count, {OpTag, OpTag});
            ({Label, OpTag, Count}) ->
+                lists:duplicate(Count, {Label, OpTag});
+           ({Label, OpTag, Count, _OptionsList}) ->
                 lists:duplicate(Count, {Label, OpTag})
         end,
-    Ops = [F(X) || X <- basho_bench_config:get(operations, [])],
+    Ops = [F(X) || X <- Operations],
     list_to_tuple(lists:flatten(Ops)).
 
+%%
+%% Expand operations list into initialized operations configuration information where used
+%%
+
+ops_configs(Id, Operations) ->
+    ops_configs(Id, Operations, []).
+
+%% Process individual operation entries in Operations list
+ops_configs(_Id, [], ACC) ->
+    ACC;
+ops_configs(Id, [{_Label, OpTag, _Count, OptionsList}|RestOps], ACC) ->
+    ops_configs(Id, RestOps, [ {OpTag, ops_configs_list(Id, OptionsList, [])} | ACC]);
+ops_configs(Id, [_Op|RestOps], ACC) ->
+    ops_configs(Id, RestOps, ACC).
+
+% Process individual config parameter entries within an Operations OptionsList
```

Comment:
'Process' is pretty generic. It looks like this function generates a keygen or valgen for each worker? Why is this needed? We still call `init_generators` yeah?